### ISSUE TITLE
Child context for shallow rendering API

### ISSUE THREAD

**Issue by DullReferenceException** [airbnb/enzyme#567]
When testing our HOC-wrapped components, we often use the pattern of shallow-rendering one more level deep:

``` es6
class MainComponent { /*...*/ };
export default connect(/*...*/)(MainComponent);
```

``` es6
const tree = shallow(<WrappedComponent/>, { context }).shallow();
```

The problem is that the child context is not being propagated, so when the inner component is rendered, its context is missing.

I see that a `childContextTypes` option was provided when doing a full render here: https://github.com/airbnb/enzyme/pull/171, but shouldn't we have the same thing for shallow rendering?

**Comment by AshCoolman** [comment]
From the same discussion:

> there is no need for it in ShallowWrapper.. (the problem was with context of deeper child nodes, there are none in ShallowWrapper)
- @jakubzitny https://github.com/airbnb/enzyme/pull/171#issuecomment-211116854

**Comment by DullReferenceException** [comment]
Yep, that was the comment, but there _is_ the need for child context if you're rendering child nodes (through calling `.shallow()` for example).

**Comment by ljharb** [comment]
Highly related to #664.

**Comment by ljharb** [comment]
Fixed in #664.
